packages/frontend/src/components/common/Filters/FilterInputs/DateFilterInputs.tsx

```diff
@@ -236,6 +239,45 @@ const DateFilterInputs: FC<FilterInputsProps<DateFilterRule>> = (props) => {
                     />
                 </MultipleInputsWrapper>
             );
+        case FilterOperator.IN_BETWEEN:
+            return (
+                <StyledDateRangeInput
+                    className={disabled ? 'disabled-filter' : ''}
+                    disabled={disabled}
+                    formatDate={(value: Date) =>
+                        formatDate(value, undefined, false)
+                    }
+                    parseDate={parseDate}
+                    value={
+                        filterRule.values
+                            ? [
+                                  new Date(filterRule.values?.[0]),
+                                  new Date(filterRule.values?.[1]),
+                              ]
+                            : [null, null]
+                    }
+                    onChange={(range: [Date | null, Date | null]) => {
+                        if (range) {
+                            onChange({
+                                ...filterRule,
+                                values: range.map((value) =>
+                                    formatDate(value, undefined, false),
+                                ),
+                            });
+                        }
```

Comment:
I suggest changing the range type to `range: [Date | null, Date | null] | null` 
Blueprint datepicker input sometimes returns null ( at least it used to ) so is good to check it is defined as you did already. 
We could also check if the values inside the array are defined and only execute the `onChange` if they are.